manager/manager.go

```diff
@@ -363,6 +363,27 @@ func (self *manager) Stop() error {
 const sysFsPCIDevicesPath = "/sys/bus/pci/devices/"
 const nvidiaVendorId = "0x10de"
 
+func setupDevices() {
+	if !detectDevices(nvidiaVendorId) {
+		glog.Info("No NVIDIA devices found.")
+		return
+	}
+
+	go func() {
+		glog.Info("Starting goroutine to initialize NVML")
+		initializeNVML()
+		if nvmlInitialized {
+			return
+		}
+		for range time.Tick(5 * time.Minute) {
+			initializeNVML()
+			if nvmlInitialized {
```

Comment:
the nvidiaDevices attribute is updated in the initializeNVML(), since it's also called somewhere else, do we need to sync lock it up when updating the nvidiaDevices?

Reply:
I don't think a lock is needed. `nvidiaDevices` is only ever written by this goroutine. Once we return from here, it's never updated. Everywhere else we are only reading the value. In the worst case, if the reads happen while `initializeNVML()` is in progress, the read will see incomplete devices. That will be corrected in the next housekeeping.

Reply:
SGTM